lib/hooks/post.ts

```diff
@@ -18,7 +29,29 @@ export function useDeletePostMutation(postId: string) {
   return useMutation(() => axios.delete("/api/post/" + postId), {
     onSuccess: () => {
       queryClient.invalidateQueries("post-" + postId);
-      router.push('/');
+      router.asPath == "/account" ? router.reload() : router.push("/");
```

Comment:
Maybe this behaviour should be controlled as part of the `mutation.mutate()` call instead. This is pretty specific behaviour for the service level. (Also we typically shouldn't need to reload the page the reflect state so see if you can avoid this by invalidating a query instead).

Reply:
Here is what I did in the changes. Let me know if this is what you had in mind: https://github.com/calebmech/Book-Bazar/blob/85-edit-post-modal/components/post-page/DeletePostForm.tsx#L38